Server/BranchingStrategy/ColorConfiguration.cs

```diff
@@ -1,13 +1,17 @@
-﻿namespace PrincipleStudios.ScaledGitApp.BranchingStrategy;
+﻿using Microsoft.Extensions.Options;
 
-public class ColorConfiguration : IColorConfiguration
+namespace PrincipleStudios.ScaledGitApp.BranchingStrategy;
+
+public class ColorConfiguration(IOptions<ColorOptions> colorOptions) : IColorConfiguration
 {
+	private readonly ColorOptions colorOptions = colorOptions.Value;
+
 	public string DetermineColor(string branchName)
 	{
-		// TODO: move to somewhere with configuration
-		if (branchName.StartsWith("rc/")) return "rgb(111, 37, 111)";
-		if (branchName.StartsWith("line/") || branchName == "main" || branchName == "master") return "rgb(111, 206, 31)";
-		if (branchName.StartsWith("integ/") || branchName.StartsWith("integrate/") || branchName.StartsWith("integration/")) return "rgb(98, 98, 98)";
-		return "rgb(55, 127, 192)";
+		if (colorOptions.Branches.TryGetValue(branchName, out string? value))
```

Comment:
Should we normalize the branch casing in either of the two usages here or are we certain they'll always match?

Reply:
I should have phrased it, "are we certain we'll get the outcome we are looking for if we have a case mismatch", but I think you answered that